buildpack.md

```diff
@@ -373,12 +386,14 @@ A buildpack's mixin requirements must be satisfied by the stack in one of the fo
 
 #### Order Resolution
 
-During detection, an order definition MUST be resolved into individual buildpack implementations.
+During detection, an order definition for image extensions (if present) and an order definition for buildpacks MUST be resolved into individual buildpack implementations.
+
+Order definitions for image extensions MUST NOT contain nested orders. If an order definition for image extensions is present, it will be pre-pended to the order definition for buildpacks (as if the order definition for image extensions were an order buildpack).
```

Comment:
```suggestion
Order definitions for image extensions MUST NOT contain nested orders.
If an order definition for image extensions is present, it MUST be prepended to the order definition for buildpacks before the resolution process occurs.
```

To make this more clear, I think we should define the following six terms early in the spec:
1. Order buildpack
2. Atomic buildpack
3. Order image extension (not currently allowed)
4. Atomic image extension
5. Order build module (currently only order buildpack)
6. Atomic build module (atomic buildpacks and image extensions)

The names are just suggestions.
Maybe it's easier to say that image extensions are buildpacks?
1. Normal order buildpack
2. Normal atomic buildpack
3. Image extension order bulidpack (not currently allowed)
4. Image extension atomic buildpack
5. Order buildpack (only normal buildpacks allowed)
6. Atomic buildpack (atomic normal and extension buildpacks)

Reply:
@sclevine what is a build module? What is the difference between an order build module and an atomic build module?

Reply:
Let's hash this out in https://github.com/buildpacks/spec/pull/300/files - we can rebase this PR when that one is merged.